### DISCUSSION TITLE
Sizing of panels/console gets messed up after resizing terminal

### DISCUSSION THREAD

**Discussion by GlenNicholls** [Textualize/rich#2116] [Q&A] [answered]
I have been using Rich's panels for quite a few applications. One issue that I've noticed is that while the application is running, resizing the terminal causes the panel/console formatting to get messed up. I've noticed that after I change the terminal size, the panel gets messed up and goes up in the console, but then the panel on the next event does indeed size properly. Is this happening because I'm not refreshing the panel/console instance enough?

**Comment by USLTD** [comment]
**Rich** doesn't support automatic resizing since it's not a **TUI** (**Text-based User Interface**) library, but **CLI** (**Command Line Interface**) library.

**Comment by GlenNicholls** [comment]
I should point out that when making the terminal larger, the Panel resizes with the window without weird borders formatted for the previous size persisting in the terminal output. However, when reducing the size the panel does correctly size properly, but the panels outlines and such of the old size do persist in the terminal if that makes sense.

Hmmm, can you expand on what you mean by Rich being a CLI instead of a TUI? The fact that there are colors, tabels, panels, etc. are TUI interfaces? Is what I'm asking what bridges the gap between Rich and Textualize in general?

**Reply by USLTD** [reply]
> Hmmm, can you expand on what you mean by Rich being a CLI instead of a TUI?

Many mistake CLI and TUI in each other or think they are same but in reality there are huge differences although it's kinda hard to draw a line between them. I mean that **rich** is CLI library because features it provides are just for making user friendly and decorative command line interface not functional interface like GUIs.

> The fact that there are colors, tabels, panels, etc. are TUI interfaces?

They are not. Those things are just used to make output more readable.

**Reply by USLTD** [reply]
See followings
https://en.m.wikipedia.org/wiki/Text-based_user_interface
https://en.m.wikipedia.org/wiki/Command-line_interface
https://askubuntu.com/questions/867416/are-there-differences-between-cli-and-tui

**Comment by willmcgugan** [comment, marked as answer]
Maybe post a video so we know we’re in the same page?

**Reply by GlenNicholls** [reply]
Here is the output. I am using the terminal in vscode, but I see identical behavior with MobaXTerm and putty as well.

![rich_discussion_2116](https://user-images.githubusercontent.com/17171913/161314299-2d1397ae-600c-4eaf-bf5c-457dfb6698d6.gif)

**Reply by willmcgugan** [reply]
Those kind of glitches are caused by the terminal wrapping output on to the next lines when the width of the terminal is reduced. It is exasperated by the trick Rich uses for the progress bars. There's nothing Rich could do to workaround that.

**Reply by GlenNicholls** [reply]
I see, thank you for the clarification!
